Update contributing information (#673)
SlimeVR/SlimeVR-Server: CONTRIBUTING.md

# Contributing to SlimeVR

This document describes essential knowledge required to contribute to the SlimeVR Server.

### Prerequisites

- [Git](https://git-scm.com/downloads)
- [Java v17+](https://adoptium.net/temurin/releases/)
- [Node.js v16+](https://nodejs.org) (We recommend the use of `nvm` instead of installing Node.js directly)

> - [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section)

Reviewer: Aside from Windows now including Webview2 by default now, I don't see any mention of the usage of `webkit2gtk` on Linux

Author: Can you propose changes. I’m not a Linux nerd.

Reviewer: ```suggestion
- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section)
- [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or ``webkit2gtk`` for Linux
```

Author: Is there a link or something that users should know about?

Reviewer: not worth it, linux is just kernel, depending on the distro they probably have different ways of installing :p

Author: Oki

> - [Windows Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or ``webkit2gtk`` for Linux

Reviewer 2: ```suggestion
- [Microsoft Edge WebView2](https://developer.microsoft.com/en-us/microsoft-edge/webview2/#download-section) or `webkit2gtk` for Linux
```

- [Rust](https://rustup.rs)

## Cloning the code
First, clone the codebase using git in a terminal in the folder you want.

```bash
git clone --recursive https://github.com/SlimeVR/SlimeVR-Server.git
```

Now you can open the codebase in [IDEA](https://www.jetbrains.com/idea/download/) (Recommended; VSCode and Eclipse also work but have limited Kotlin support).

## Building the code

### Java (server)

The Java code is built with `gradle`, a CLI tool that manages java projects and their
dependencies.
- You can run the server by running `./gradlew run` in your IDE's terminal.
- To compile the code, run `./gradlew shadowJar`. The result will
be at `server/build/libs/slimevr.jar` (you can ignore `server.jar`).

> (Note: your IDE may do all of the above for you.)

Reviewer 2: ```suggestion
(Note: Your IDE may be able to do all of the above for you.)
```

### Tauri (gui)

- Run `npm install` in your IDE's terminal to download and install dependencies.

> - To launch the GUI in dev mode, run `npm run gui`.

Reviewer 2: ```suggestion
- To launch the GUI in dev mode, run `npm run tauri dev`.
```
Shouldn't this be like this instead? idk what `npm run gui` does.

Author: `npm run gui` runs `npm i solarxr-protocol`, so it's better to avoid confusion and extra steps.

- Finally, to compile for production, run `npm run tauri build`. The result
will be at `target/release/slimevr.exe`.

## Code style

### Java (server)

The Java code is auto-formatted with [spotless](https://github.com/diffplug/spotless/tree/main/plugin-gradle).
Code is checked for autoformatting whenever you build, but you can also run
`./gradlew spotlessCheck` if you prefer.

> To auto-format your Java code from the command line, you can run `./gradlew spotlessApply`.

Reviewer 2: ```suggestion
To auto-format your Java and Kotlin code from the command line, you can run `./gradlew spotlessApply`.
```

We recommend installing support for spotless in your IDE, and formatting
whenever you save a file to make things easy.

If you need to prevent autoformatting for a select region of code, use
`// @formatter:off` and `// @formatter:on`

#### Setting up spotless for IntelliJ IDEA

> * Install https://plugins.jetbrains.com/plugin/18321-spotless-gradle.

Reviewer 2: ```suggestion
* Install https://plugins.jetbrains.com/plugin/18321-spotless-gradle
```

Eclipse will only do a subset of the checks in `spotless`, so you may still want to do
`./gradlew spotlessApply` if you ever see an error from spotless.

### Tauri (gui)

We use ESLint and Prettier to format GUI code.
- First, go into the GUI's directory with your terminal by running `cd gui`.
- To check code formatting, run `npm run lint`.
- To fix code formatting, run `npm run lint:fix` and `npm run format`

> Don't forget to run `cd ..` to go back in the main directory.

Reviewer 2: ```suggestion
Don't forget to run `cd ..` to return to the root directory.
```
